site/content/how-to/traffic-management/tls-passthrough.md

```diff
@@ -0,0 +1,231 @@
+
+Run the following command to verify the resources were created:
+
+```shell
+kubectl get pods,svc
+```
+
+The output should include the **secure-app** pod and the **secure-app** service:
+
+```text
+NAME                              READY   STATUS      RESTARTS   AGE
+pod/secure-app-575785644-kzqf6    1/1     Running     0          12s
+
+NAME                  TYPE        CLUSTER-IP        EXTERNAL-IP   PORT(S)    AGE
+service/secure-app    ClusterIP   192.168.194.152   <none>        8443/TCP   12s
+```
+
+Create a gateway. This will create a TLS listener with the hostname `*.example.com` and a TLS mode of passthrough. Copy and paste this into your terminal.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: default
+spec:
+  gatewayClassName: nginx
+  listeners:
+  - name: tls
+    port: 443
+    protocol: TLS
+    hostname: "*.example.com"
+    allowedRoutes:
+      namespaces:
+        from: All
+      kinds:
+        - kind: TLSRoute
+    tls:
+      mode: Passthrough
+EOF
+```
+
+This gateway will configure NGINX Gateway Fabric to accept TLS connections on port 443 and route them to the corresponding backend services without decryption. The routing is done based on the SNI, which allows clients to specify a server name (like example.com) during the SSL handshake.
+
+{{< note >}} It is possible to add an HTTPS listener on the same port that terminates TLS connections so long as the hostname does not overlap with the TLS listener hostname. {{< /note >}}
+
+Create a TLSRoute that attaches to the gateway and routes requests to `app.example.com` to the `secure-app` service:
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSRoute
+metadata:
+  name: tls-secure-app-route
+  namespace: default
+spec:
+  parentRefs:
+  - name: gateway
+    namespace: default
+  hostnames:
+  - "app.example.com"
+  rules:
+  - backendRefs:
+    - name: secure-app
+      port: 8443
+EOF
+```
+
+{{< note >}}To route to a service in a namespace different from the TLSRoute namespace, create a [ReferenceGrant](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1beta1.ReferenceGrant) to permit the cross-namespace reference. {{< /note >}}
+
+## Send traffic
+
+Using the external IP address and port for NGINX Gateway Fabric, send traffic to the `secure-app` application.
+
+{{< note >}}If you have a DNS record allocated for `app.example.com`, you can send the request directly to that hostname, without needing to resolve.{{< /note >}}
+
+Send a request to the `secure-app` service on the TLS port with the `--insecure` flag. The `--insecure` flag is required because the `secure-app` is using self-signed certificates.
```

Comment:
```suggestion
Send a request to the `secure-app` Service on the TLS port with the `--insecure` flag. The `--insecure` flag is required because the `secure-app` is using self-signed certificates.
```